taxi/tests/test_views.py

```diff
@@ -0,0 +1,197 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_URL = reverse("taxi:car-list")
+DRIVER_URL = reverse("taxi:driver-list")
+
+
+class PublicManufacturerTests(TestCase):
+    def test_login_required(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="test123",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="Audi")
+        Manufacturer.objects.create(name="Mercedes")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers),
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+
+class ManufacturerSearchTests(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="test123",
+        )
+        self.client.force_login(self.user)
+
+    def test_search_manufacturer(self):
+        Manufacturer.objects.create(name="Audi")
+        Manufacturer.objects.create(name="Mercedes")
+        Manufacturer.objects.create(name="Toyota")
```

Comment:
To avoid the code duplication, place this part in the `setUp`. The same relates to the `DriverSearchTests` as well